dist/filepond-plugin-image-overlay.esm.js

```diff
@@ -61,11 +61,8 @@ const createFullSizeOverlay = (item) => {
 
   determineImageOverlaySize(imgUrl, imgContainer);
 
-  let body = document.getElementsByTagName('body')[0];
-
   overlay.appendChild(imgContainer);
-  body.appendChild(overlay);
-
+  document.getElementById(`filepond--item-${item.id}`).appendChild(overlay);
```

Comment:
appending overlay in file item list _div_ instead of the body tag, so that the overlay can be shown over the image.